client/sites-dashboard/components/sites-table-row.tsx

```diff
@@ -91,19 +100,26 @@ export default function SitesTableRow( { site }: SiteTableRowProps ) {
 						<SiteIcon siteId={ site.ID } size={ 50 } />
 					</a>
 					<div style={ { marginLeft: '20px' } }>
-						<SiteName>
-							<a href={ getDashboardUrl( site.slug ) } title={ __( 'Visit Dashboard' ) }>
-								{ site.name ? site.name : __( '(No Site Title)' ) }
-							</a>
-						</SiteName>
-						<SiteUrl
-							href={ site.URL }
-							target="_blank"
-							rel="noreferrer"
-							title={ __( 'Visit Site' ) }
-						>
-							{ displaySiteUrl( site.URL ) }
-						</SiteUrl>
+						<div style={ { display: 'flex', alignItems: 'center', marginBottom: '8px' } }>
+							<SiteName style={ { marginRight: '8px' } }>
+								<a href={ getDashboardUrl( site.slug ) } title={ __( 'Visit Dashboard' ) }>
+									{ site.name ? site.name : __( '(No Site Title)' ) }
+								</a>
+							</SiteName>
+							{ P2Badge( site ) }
```

Comment:
> How about we turn this into a real element by using `<P2Badge site={site} />`?

@zaguiini Could you clarify what you mean?

Reply:
So `P2Badge` should either be:
1. a fully-fledged "component" in React speak, that could hypothetically have its own state and lifecycle events, or
2. just a helper function that decides whether to return a `<SitesP2Badge>` or not.

My Google foo could be off, but I struggled to find a blog post that did a good job at describing the trade-offs between the two. I'm probably missing something here but if something's a component it will:
- Have access to component lifecycle events e.g. mounting, prop updated, unmount, etc.
- Be able to call hooks
- Appear as a component in the React dev tools
- Participate in [reconciliation](https://reactjs.org/docs/reconciliation.html) (so there are potential performance considerations)

UI that's rendered by a helper function is totally indistinguishable from UI that's rendered directly in the render function itself.

A colleague of mine once said that if it's going to be updated more than it's mounted then go with (1), if it's going to be mounted more than it's updated then go with (2). I think that advise holds strongest situations where performance is important, like 100s of rows in a table.

So practically, is this case, if you decide to go with:
1. then you should keep `P2Badge` capitalised, change the function arguments so it accepts a props object, and invoke it like `<P2Badge />` instead of `{ P2Badge() }`
2. then you should lowercase the function to `p2Badge` and leave the rest as is (or maybe even `renderP2Badge` since sometimes rendering helper functions are prefixed with `render*`)

Reply:
IMO I think (2) is better in this case since `p2Badge()` isn't so much a component as just a helper for tidying up the is-this-a-p2 logic.

Reply:
@p-jackson This is a great explanation, thank you.

@zaguiini talked me through it so I properly understood it 😁 Applied changes with 29cfa9fbcc